src/adapter/src/coord/sequencer.rs

```diff
@@ -230,6 +230,40 @@ impl<S: Append + 'static> Coordinator<S> {
                 tx.send(self.sequence_show_variable(&session, plan), session);
             }
             Plan::SetVariable(plan) => {
+                // Add notice if database or cluster value does not correspond to a catalog item.
+                if plan.name == "database" || plan.name == "cluster" {
+                    let value = match plan.value.clone() {
+                        mz_sql::ast::SetVariableValue::Ident(ident) => Some(ident.into_string()),
+                        _ => None,
+                    };
+                    match plan.name.as_str() {
+                        "database" => {
+                            if value.is_none()
+                                || matches!(
+                                    self.catalog.resolve_database(&value.unwrap()),
+                                    Err(CatalogError::UnknownDatabase(_))
+                                )
+                            {
+                                session.add_notice(AdapterNotice::DatabaseDoesNotExist {
+                                    name: format!("{}", plan.value),
+                                });
+                            }
+                        }
+                        "cluster" => {
+                            if value.is_none()
+                                || matches!(
+                                    self.catalog.resolve_compute_instance(&value.unwrap()),
+                                    Err(CatalogError::UnknownComputeInstance(_))
+                                )
+                            {
+                                session.add_notice(AdapterNotice::ClusterDoesNotExist {
+                                    name: format!("{}", plan.value),
+                                });
+                            }
+                        }
+                        &_ => {}
+                    }
+                }
```

Comment:
I thiiink you want to push all this logic into `sequence_set_variable`? Unless there's some reason I'm not seeing that you didn't do that on purpose!